File: dongyimaisellergoodsinterface/src/main/java/com/dongyimai/service/BrandService.java
package com.dongyimai.service;

import com.dongyimai.entity.PageResult;
import com.dongyimai.entity.Result;
import com.dongyimai.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    public List<TbBrand> findAll();

    //返回分页列表
    public PageResult findPage(int pageNum,int pageSize);

    public void add(TbBrand tbBrand);
    public void update(TbBrand tbBrand);

    /**根据ID获取实体
     * @param id
     * @return
     */
    public TbBrand findOne(long id);

    /**批量删除
     * @param ids
     */
    public void delete(long[] ids);

    /**根据条件分页查询
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(TbBrand tbBrand,int pageNum,int pageSize);

    /**品牌下拉框数据
     * @return
     */
    List<Map> selectOptionList();
}

File: dongyimaimanagerweb/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {
    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    };
    this.findPage=function (page,rows) {
        return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
    };
    this.add=function (entity) {
        return $http.post('../brand/add.do',entity);
    };
    this.update=function (entity) {
        return $http.post('../brand/update.do',entity);
    }
    this.delete=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    };
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    };
    this.search=function (searchEntity,page,rows) {
        return $http.post('../brand/search.do?page='+page+'&rows='+rows,searchEntity);
    }

    //下拉列表数据
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do')
    }
});